### PR TITLE
MINIFICPP-440 Added LibreSSL as an external project build, statically…

### PR DESCRIPTION
**Pull request by achristianson**
…-linked for cases when it isn't available in the target environment for dynamic linking

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by achristianson** [comment]
Looking into potential Macintosh issue...

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,8 +25,10 @@ set(PROJECT_VERSION_MINOR 4)
 set(PROJECT_VERSION_PATCH 0)
 option(SKIP_TESTS "Skips building all tests." OFF)
 option(PORTABLE "Instructs the compiler to remove architecture specific optimizations" ON)
+option(USE_SYSTEM_OPENSSL "Instructs the build system to search for and use an SSL library available in the host system" OFF)
```
Comment (phrocker) [inline CMakeLists.txt]: I'm not in favor of changing the system dependency in a minor release. I would be in favor of moving to USE_SYSTEM_OPENSSL as the default until the next major release.

Comment (achristianson) [reply CMakeLists.txt]: OK I'll invert that default. The thinking is that building/static-linking makes the binary more portable and that's probably the config we want for the release binary. But I agree it's a fairly significant change for a minor release.

**Comment by achristianson** [comment]
@phrocker I inverted the default to the more conservative option for 0.5.0. Would 0.6.0 be the target for switching it back? I created MINIFICPP-441 to track and will set the release of that ticket accordingly.

**Comment by phrocker** [comment]
The travis failure is not related. will merge. Thanks

**Pull request closed without merging**